SharedSources/VLCMediaLibraryManager.swift

```diff
@@ -79,21 +79,19 @@ class VLCMediaLibraryManager: NSObject {
     private static let migrationKey: String = "MigratedToVLCMediaLibraryKit"
 
     private var didMigrate = UserDefaults.standard.bool(forKey: VLCMediaLibraryManager.migrationKey)
-
+    private var didFinishDiscovery = false
     // Using ObjectIdentifier to avoid duplication and facilitate
     // identification of observing object
     private var observers = [ObjectIdentifier: Observer]()
 
-    private lazy var medialib: VLCMediaLibrary = {
-        let medialibrary = VLCMediaLibrary()
-        medialibrary.delegate = self
-        return medialibrary
-    }()
+    private var medialib: VLCMediaLibrary!
 
     weak var migrationDelegate: MediaLibraryMigrationDelegate?
 
     override init() {
         super.init()
+        medialib = VLCMediaLibrary()
+        medialib.delegate = self
```

Comment:
Instead of implicitly unwrapping `medialib`, it could be initialized.

Additionally, I would consider putting `medialib.delegate` inside `setupMediaLibrary()`.